frontend/packages/ceph-storage-plugin/integration-tests/views/installFlow.view.ts

```diff
@@ -67,54 +106,42 @@ export const goToWorkLoads = async () => {
   await browser.wait(until.and(crudView.untilNoLoadersPresent));
   await sideNavView.clickNavLink(['Workloads', 'Pods']);
   await browser.wait(until.and(crudView.untilNoLoadersPresent));
-  await browser.wait(until.visibilityOf(namespaceDropdown));
+  await browser.wait(until.visibilityOf(currentSelectors.namespaceDropdown));
   await $('.co-namespace-selector button').click();
-  await browser.wait(until.elementToBeClickable(openshiftStorageItem));
+  await browser.wait(until.elementToBeClickable(currentSelectors.openshiftStorageItem));
   await $('#openshift-storage-link').click();
   await browser.wait(until.and(crudView.untilNoLoadersPresent));
 };
 
 // Operators page
 export const selectWorkerRows = async () => {
-  // Wait for 3 inputs to show up
   const selectedNodes = [];
-  await browser.wait(until.presenceOf($('[data-label="Role"]')), 10000);
-  await browser.sleep(5 * SECOND);
-  expect(await browser.wait($$('tbody tr').count())).toBeGreaterThanOrEqual(3);
-  const isAllSeleted = await selectAllBtn.isSelected();
-  if (isAllSeleted === true) await selectAllBtn.click();
-  await browser.wait(until.not(until.elementToBeSelected(selectAllBtn)));
-  const workerAz = [];
-  const table = $('.pf-c-table.pf-m-compact tbody');
-  const rows = table.$$('tr');
+  const isAllSeleted = await currentSelectors.selectAllBtn.isSelected();
+  if (isAllSeleted === true) await click(currentSelectors.selectAllBtn);
+  const workersAZ = [];
+  const table = $('[aria-label="Node Table"]');
+  const rows = table.$$('tbody tr');
   rows.each(async (row, index) => {
-    const currRow = JSON.stringify(await row.getText());
-    const newCurrRow = currRow.replace(/\\n/g, ' ');
-    let splitedRow = [];
-    splitedRow = newCurrRow.split(/\s/g);
-    if (splitedRow[3] === 'worker') {
-      workerAz.push(splitedRow[4]);
-      if (
-        !(await $$('tbody input')
-          .get(index)
-          .isSelected())
-      ) {
-        await browser.sleep(1 * SECOND);
-        await $$('tbody input')
+    await (async () => {
+      let splitedRow = [];
+
+      const currRow = JSON.stringify(await row.getText());
+      const newCurrRow = currRow.replace(/\\n/g, ' ');
+      splitedRow = newCurrRow.split(/\s/g);
+      if (selectedNodes.length < 3) {
+        workersAZ.push(splitedRow[4]);
+        const checkBox = await table.$$('tbody input').get(index);
+        const nodeName = await table
+          .$$('tbody a')
           .get(index)
-          .click();
-        await browser.wait(until.elementToBeSelected($$('tbody input').get(index)));
-        selectedNodes.push(
-          await table
-            .$$('tbody a')
-            .get(index)
-            .getText(),
-        );
+          .getText();
+        await click(checkBox);
+        await browser.wait(until.elementToBeSelected(checkBox));
+        selectedNodes.push(nodeName);
```

Comment:
Can you explain what are you checking here? Are you individually selecting nodes and not by select all nodes?

Reply:
On second though selecting all at once would be much easier. Thanks will look into it.